hail/python/hail/vds/impex.py

```diff
@@ -0,0 +1,335 @@
+    Export to VCF as a block-compressed file:
+
+    >>> hl.vds.export_vcf(dataset, 'output/example.svcr.vcf.bgz')  # doctest: +SKIP
+
+    Notes
+    -----
+    While the innovations present in VDS/SVCR have been standardized in the
+    `VCF 4.5 spec <https://samtools.github.io/hts-specs/VCFv4.5.pdf>`__, at current, the library
+    that we use to parse VCF headers does not support VCF 4.5, so we instead output a standard,
+    widely supported version 4.2 VCF like :func:`hail.export_vcf`. We expect this to change in the
+    future as support for VCF 4.5 improves.
+
+    All recommendations of :func:`hail.export_vcf` apply to this method, and the arguments are
+    identical. Please refer to the documentation of that method as all recommendations and warnings
+    apply to this method as well.
+
+    Note
+    ----
+    Some data transformations take place before export.
+
+    #. If the local alleles field, ``LA`` is present, it is converted to the standardized local
+       alternate alleles field, ``LAA``.
+    #. Both of ``LGT``, and ``LPGT`` are converted to ``GT`` and ``PGT``, using
+       :func:`.vds.lgt_to_gt`, if necessary.
+    #. The ``gvcf_info`` field produced by the :class:`hail.vds.combiner.VariantDatasetCombiner` is
+       dropped. VCF does not support ``struct`` type entry fields. To preserve this data, flatten
+       the struct onto the entries::
+
+            flat_gvcf_info = {f'gi_{name}': vds.variant_data.gvcf_info[name] for name in vds.variant_data.gvcf_info}
+            vds.variant_data = vds.variant_data.transmute_entries(**flat_gvcf_info)
+
+    Parameters
+    ----------
+    dataset : :class:`.MatrixTable`
+        Dataset.
+    output : :class:`str`
+        Path of .vcf or .vcf.bgz file to write.
+    append_to_header : :class:`str`, optional
+        Path of file to append to VCF header.
+    parallel : :class:`str`, optional
+        If ``'header_per_shard'``, return a set of VCF files (one per
+        partition) rather than serially concatenating these files. If
+        ``'separate_header'``, return a separate VCF header file and a set of
+        VCF files (one per partition) without the header. If ``None``,
+        concatenate the header and all partitions into one VCF file.
+    metadata : :obj:`dict` [:obj:`str`, :obj:`dict` [:obj:`str`, :obj:`dict` [:obj:`str`, :obj:`str`]]], optional
+        Dictionary with information to fill in the VCF header. See
+        :func:`get_vcf_metadata` for how this
+        dictionary should be structured.
+    tabix : :obj:`bool`, optional
+        If true, writes a tabix index for the output VCF.
+        **Note**: This feature is experimental, and the interface and defaults
+        may change in future versions.
+    """
+    ref, var = dataset.reference_data, dataset.variant_data
+
+    # fix up ref data
+    if 'END' in ref.entry:
+        ref = ref.drop('END')
+
+    if 'gvcf_info' in var.entry and isinstance(var.gvcf_info.dtype, tstruct):
+        warning(
```

Comment:
Do you need to warn here? You've documented this above and given an example for how to preserve this information.

I view warnings as a way of telling users about changing behaviour or when they've been a bit naughty (but not terrible) and this doesn't really fall into that. What's your view?